sys/mips/boot.c

```diff
@@ -100,11 +102,20 @@ __boot_text void *mips_init(void) {
     pte[PTE_INDEX(va)] = PTE_PFN(pa) | PTE_KERNEL;
 
 #if KASAN /* Prepare KASAN shadow mappings */
+  size_t kasan_sanitized_size =
+    roundup(va - MIPS_PHYS_TO_KSEG2(text), SUPERPAGESIZE);
+  /*
+   * TODO(pj): I believe that it's a correct value here... but thread0 stack
+   * shadow check triggers BIG error in that case so let's create bigger shadow
+   * for now.
+   * kasan_sanitized_size >> KASAN_SHADOW_SCALE_SHIFT;
+   */
+  size_t kasan_shadow_size = kasan_sanitized_size;
```

Comment:
I think you're getting an error because `kasan_shadow_size` isn't aligned to `SUPERPAGESIZE`, so `num_pde = kasan_shadow_size / SUPERPAGESIZE` will round down, and we will map less pages than required. The way to fix this is `kasan_sanitized_size = roundup(va - MIPS_PHYS_TO_KSEG2(text), SUPERPAGESIZE << KASAN_SHADOW_SCALE_SHIFT)`

Reply:
Oh, thanks!